x-pack/plugin/src/test/java/org/elasticsearch/xpack/test/rest/XPackRestIT.java

```diff
@@ -229,11 +230,14 @@ private void awaitCallApi(String apiName,
                               CheckedFunction<ClientYamlTestResponse, Boolean, IOException> success,
                               Supplier<String> error) {
         try {
-            // The actual method call that sends the API requests returns a Future, but we immediately
-            // call .get() on it so there's no need for this method to do any other awaiting.
-            ClientYamlTestResponse response = callApi(apiName, params, bodies, getApiCallHeaders());
-            assertEquals(response.getStatusCode(), HttpStatus.SC_OK);
-            success.apply(response);
+            final AtomicReference<ClientYamlTestResponse> response = new AtomicReference<>();
+            assertBusy(() -> {
+                // The actual method call that sends the API requests returns a Future, but we immediately
+                // call .get() on it so there's no need for this method to do any other awaiting.
+                response.set(callApi(apiName, params, bodies, getApiCallHeaders()));
+                assertEquals(HttpStatus.SC_OK, response.get().getStatusCode());
+            });
+            success.apply(response.get());
```

Comment:
Would it work to just do the following, since `assertBusy` only re-runs the block if it throws an exception?

```java
assertBusy(() -> {
    // The actual method call that sends the API requests returns a Future, but we immediately
    // call .get() on it so there's no need for this method to do any other awaiting.
    ClientYamlTestResponse response = callApi(apiName, params, bodies, getApiCallHeaders());
    assertEquals(HttpStatus.SC_OK, response.get().getStatusCode());
    success.apply(response);
});
```

Reply:
I considered that, but I didn't want to swallow any assertion errors thrown by `success.apply()`, which today may call `ObjectPath#evaluate` which could well assert things in future even if it doesn't today.